x/ccv/child/keeper/genesis.go

```diff
@@ -38,6 +43,31 @@ func (k Keeper) InitGenesis(ctx sdk.Context, state *types.GenesisState) {
 		// set parent client id.
 		k.SetParentClient(ctx, clientID)
 	} else {
+		// verify that latest consensus state on parent client matches the initial validator set of restarted chain
+		// thus, IBC genesis MUST run before CCV child genesis
+		consState, ok := k.clientKeeper.GetLatestClientConsensusState(ctx, state.ParentClientId)
+		if !ok {
+			panic("consensus state for parent client not found. MUST run IBC genesis before CCV child genesis")
+		}
+		tmConsState, ok := consState.(*ibctmtypes.ConsensusState)
+		if !ok {
+			panic(fmt.Sprintf("consensus state has wrong type. expected: %T, got: %T", &ibctmtypes.ConsensusState{}, consState))
+		}
+
+		// ensure that initial validator set is same as initial consensus state on provider client.
+		// this will be verified by provider module on channel handshake.
+		vals, err := tmtypes.PB2TM.ValidatorUpdates(state.InitialValSet)
+		if err != nil {
+			panic(err)
+		}
+		valSet := tmtypes.NewValidatorSet(vals)
+
+		if !bytes.Equal(tmConsState.NextValidatorsHash, valSet.Hash()) {
```

Comment:
This check requires that the initial validator set in the GenesisState matches the one from the latest consensus state stored in the provider client. This consensus state depends on the last ClientUpdate before the GenesisState was exported. How can an off-chain process identify the validator set of that consensus state?